[CI] AccountableQueryCircuitBreakerIT testRangeQueryMemoryReleased failing

**Build Scans:**
- [elasticsearch-periodic #12148 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/soledx373jrg4)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.indices.memory.breaker.AccountableQueryCircuitBreakerIT.testRangeQueryMemoryReleased" -Dtests.seed=D9C982335E877A8A -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=ku-Latn-TR -Dtests.timezone=Australia/Canberra -Druntime.java=26
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.indices.memory.breaker.AccountableQueryCircuitBreakerIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRangeQueryMemoryReleased'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request breaker memory should be released after search completes expected:<131520> but was:<0>
```

**Issue Reasons:**
- [8.19] 1 consecutive failures in test testRangeQueryMemoryReleased
- [8.19] 1 consecutive failures in step single-processor-node-tests
- [8.19] 1 consecutive failures in pipeline elasticsearch-periodic

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/AccountableQueryCircuitBreakerIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/AccountableQueryCircuitBreakerIT.java
@@ -216,7 +216,9 @@ public class AccountableQueryCircuitBreakerIT extends ESIntegTestCase {
     private void assertQueryMemoryReleased(QueryBuilder query) throws Exception {
         long baseline = getRequestBreakerEstimated();
         client().prepareSearch(INDEX_NAME).setQuery(query).get().decRef();
-        long estimated = getRequestBreakerEstimated();
-        assertEquals("Request breaker memory should be released after search completes", baseline, estimated);
+        assertBusy(() -> {
+            long estimated = getRequestBreakerEstimated();
+            assertEquals("Request breaker memory should be released after search completes", baseline, estimated);
+        });
     }
 }

